src/accelerate/utils/modeling.py

```diff
@@ -124,11 +124,14 @@ def set_module_tensor_to_device(
             new_value = value.to(device)
         else:
             new_value = torch.tensor(value, device=device)
-    if is_buffer:
-        module._buffers[tensor_name] = new_value
-    else:
-        new_value = nn.Parameter(new_value, requires_grad=old_value.requires_grad)
-        module._parameters[tensor_name] = new_value
+
+        if is_buffer:
+            module._buffers[tensor_name] = new_value
+        elif module._parameters[tensor_name].device.type != "cuda":
+            param_cls = type(module._parameters[tensor_name])
+            kwargs = module._parameters[tensor_name].__dict__
+            new_value = param_cls(new_value, requires_grad=old_value.requires_grad, **kwargs).to(device)
+            module._parameters[tensor_name] = new_value
```

Comment:
This should stay at the previous level of indentation. To make sure there is absolutely zero breaking change, how about doing an `else`  with just `module._parameters[tensor_name].to(device)` ?

Reply:
Just added it and seems to work fine! 
Can we wait a bit before merging the PR?

Reply:
Great thanks a lot!!